### PR TITLE
[KOGITO-6524] Replace hardcoded version with placeholders in custom dashboards.

### PR DESCRIPTION
**Pull request by gitgabrio**
@danielezonca @radtriste 

This PR
1) replace hardcoded version with placeholders inside custom dashboards json
2) remove version from  custom dashboards file names
3) move custom dashboards files inside src/main/filtered-resources
4) modify pom so that custom dashboards files are filtered and moved to src/main/resources at `process-resources`phase
5) add `src/main/resources/META-INF/dashboards` to .gitignore

The needs of the slightly contrived resource management is because test run before package and look for files inside src/main/resources. So I had to configure maven to filter and move them before test execution.

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

**WARNING! Please make sure you are opening your PR against `main` branch!**

- [ ] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

File: kogito-springboot-examples/dmn-drools-springboot-metrics/pom.xml [hunk 1]
```diff
@@ -108,6 +108,22 @@
...
         <artifactId>maven-resources-plugin</artifactId>
         <executions>
+          <execution>
+            <id>copy-filtered-resources</id>
+            <phase>process-resources</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.basedir}/src/main/resources/META-INF/dashboards</outputDirectory>
```
Comment (radtriste) [inline kogito-springboot-examples/dmn-drools-springboot-metrics/pom.xml:122]: Could we output this into the target folder ?

Comment (gitgabrio) [reply kogito-springboot-examples/dmn-drools-springboot-metrics/pom.xml:122]: @radtriste 
I add explanation in the desciption of this PR. Anyway, the reason is that kogito code (outside my modification) is looking for resources inside `src/main/resources`, and tests run before package. Anyway, during package, they will be copied to `target`.

Comment (radtriste) [reply kogito-springboot-examples/dmn-drools-springboot-metrics/pom.xml:122]: sorry I missed that

**Review by radtriste — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -53,3 +53,5 @@ kogito-travel-agency/extended/scripts/persistence/
 
 #Grafana generated dashboards
 **/docker-compose/grafana/provisioning/dashboards/*.json
+/kogito-springboot-examples/dmn-drools-springboot-metrics/src/main/resources/META-INF/dashboards/
+/kogito-quarkus-examples/dmn-drools-quarkus-metrics/src/main/resources/META-INF/dashboards/
```
Comment (danielezonca) [inline .gitignore]: Please move this ignore locally to the module to help future refactoring and similar

**Review by danielezonca — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (gitgabrio) [reply .gitignore]: @danielezonca 
done

**Comment by gitgabrio** [comment]
@danielezonca @radtriste @evacchi 
I had to change the quarkus example setup so that filtered dashboards are directly copied to target.
During debug, I noticed that `KogitoBuildContext` contains very different paths between quarkus (-> `target`) and springboot (-> `src`). That's why quarkus did not pick files from src/main/resources during test execution.
What it is strange for me to understand, is that the original PR (https://github.com/kiegroup/kogito-examples/pull/1003) successfully built, and in that PR dashboards where put directly inside src/main/resources. 
The only difference with the current one is that the files are moved at "process-resource" phase, that should happen before anything else.

**Pull request merged**